python/pyarrow/_dataset.pyx

```diff
@@ -400,6 +438,34 @@ cdef class Fragment:
         """
         return Expression.wrap(self.fragment.partition_expression())
 
+    def scan(self, MemoryPool memory_pool=None):
```

Comment:
Should we be able to pass a columns/filter arguments here? (similar as Dataset.scan and Scanner)

Reply:
As above, these options are already specified in `Dataset.get_fragments()`. I could add optional parameters which would refine the filter/projection further

Reply:
Ah, yes, they are specified there indeed, if creating from a Dataset. But when re-creating a new fragment from its parts, it's still useful to specify filter/columns

Reply:
Fragment creation can be accomplished with `FileFormat.make_fragment`, which *does* take filter/columns. Please compare this method to `Scanner.scan` rather than `Dataset.scan`. I'm opposed to putting any more parameters here; they're already specifiable elsehwere

Reply:
I don't think that's desired that fragment store the filter, is that a temporary solution until ARROW-8065?

Reply:
Yes, I think most of the discussion / disagreement is indeed to bring back to that issue. 

I find it strange to tie the filter/columns to fragment creation, as for me conceptually that are scanning options. And if we decide to remove ScanOptions from the fragment creation, that would indeed entail that?